src/components/DetailsView/DetailsView.js

```diff
@@ -1,59 +1,42 @@
-import React, { Component, createRef } from 'react';
+import { useRef, useState } from 'react';
 import ReactDOM from 'react-dom';
+import { usePokemons } from '../../context/PokemonsProvider';
 import BackButton from '../BackButton/BackButton';
 import { Card } from '../Card';
 import { Details } from '../Details';
 import { Overlay } from '../Overlay';
 import './DetailsView.css';
 
-class DetailsView extends Component {
-	constructor( props ) {
-		super( props );
-		this.detailsViewRef = createRef( null );
+export default function DetailsView() {
+	const detailsViewRef = useRef( null );
+	const { setCurrentPokemonId, currentPokemon } = usePokemons();
```

Comment:
I would switch the order just for the habit that the data/state comes first and the function comes after.

Reply:
Interesting... Done